tree-sitter-core.el

```diff
@@ -3,14 +3,17 @@
 ;; Copyright (C) 2019  Tuấn-Anh Nguyễn
 ;;
 ;; Author: Tuấn-Anh Nguyễn <ubolonton@gmail.com>
+;;         Jorge Javier Araya Navarro <jorgejavieran@yahoo.com.mx>
 
 ;;; Commentary:
 
 ;; This file contains the core functionalities of tree-sitter.
 ;;
 ;;; Code:
 
-(require 'tree-sitter-dyn)
+(if (functionp 'module-load)
+    (require 'tree-sitter-dyn nil t)
```

Comment:
I think the check should be in a separate top-level form. Code analysis tools  work better with top-level `require`.

Also, `require` shouldn't be used with `NOERROR` flag set, as we still want to fail on other errors.

Reply:
roger roger